core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -4577,6 +4577,10 @@ private Optional<TableVersion> extractTableVersion(Table table, QualifiedObjectN
             }
 
             if (!metadata.isValidTableVersion(session, tableName, extractedVersion)) {
```

Comment:
I would argue that this check (`isValidTableVersion`) should be done  internally in the `ConnectorMetadata` implementation while retrieving the table handle.

See `io.trino.plugin.iceberg.IcebergMetadata#getTableHandle(io.trino.spi.connector.ConnectorSession, io.trino.spi.connector.SchemaTableName, java.util.Optional<io.trino.spi.connector.ConnectorTableVersion>, java.util.Optional<io.trino.spi.connector.ConnectorTableVersion>)`

Reply:
I was trying to avoid that, but I think you're right.
There are 4 potential problems when querying a table `FOR VERSION`:

1. table doesn't exist
2. connector doesn't support versioning at all
3. given version type is not supported (e.g. versions are `bigint` (as in Iceberg) but user provided a `varchar`)
4. version type is fine, but requested version doesn't exist (e.g. bad Iceberg snapshot id)

If we rely on `isValidTableVersion` + `getTableHandle` to check table existence, we cannot differentiate between 2. 3 and 4. And honestly, we don't have to, since the connector will do such checks anyway.

Therefore I am removing `isValidTableVersion`.

cc @martint @jacob-stein